### PR TITLE
Vendor go dependencies

### PR DESCRIPTION
**Pull request by pajakd**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
I ran `go mod vendor` in the main directory. I would like to verify if having the dependencies vendored shortens build time (to help decide in the discussion #2374).

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note 
NONE
```

### CODE REVIEW

**Comment by alculquicondor** [comment]
It looks like commands like this:

https://github.com/kubernetes-sigs/kueue/blob/6ecaa877ee65791e6333f147fafcbb7888039a03/Makefile-deps.mk#L99C20-L99C84

Are not pointing to the vendor folder

**Comment by pajakd** [comment]
Yes, this command returns an empty string if the vendor directory is present. 

What is more problematic is that `/vendor` directory only contains Go-packages (https://github.com/golang/go/issues/26366). So, the files (the CRD .yaml definitions) that we are trying to copy here are not in the vendored package directory.

**Comment by alculquicondor** [comment]
Re: make verify

We probably need to change the verify rules to exclude the vendor folder. It might be formatting or doing similar things.

File: Dockerfile [hunk 1]
```diff
@@ -10,9 +10,6 @@ WORKDIR /workspace
 # Copy the Go Modules manifests
 COPY go.mod go.mod
 COPY go.sum go.sum
```
Comment (alculquicondor) [inline Dockerfile:12]: I think we can remove these 2 lines too

**Comment by alculquicondor** [comment]
Also need to update https://github.com/kubernetes-sigs/kueue/blob/main/cmd/importer/Dockerfile

**Comment by pajakd** [comment]
"There are no nodes that your pod can schedule" -- looks like too many build jobs ran in parallel.

File: Dockerfile (hunk 1 above)
Comment (pajakd) [reply Dockerfile]: Done.

**Comment by pajakd** [comment]
Fixed the problems with verify (had to exclude /vendor directory from `gofmt`).

It looks like the speedup in build time is small (or none). 

The only benefit from vendored dependencies would be more hermetic build.

**Comment by alculquicondor** [comment]
/test pull-kueue-build-image-main

(curious if it's consistently faster)

**Comment by alculquicondor** [comment]
It looks like the builds are between 20 to 30s faster.

@tenzen-y wdyt?

**Comment by alculquicondor** [comment]
I think we also need a change in `.github/dependabot.yaml` https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file#vendor

**Comment by pajakd** [comment]
As far as I understand with our configuration (`package-ecosystem: "gomod"`)  we don't need to change anything in our configuration. From the documentation "Don't use this option if you're using gomod as Dependabot automatically detects vendoring for this tool.". Additional reference: https://github.blog/changelog/2020-10-19-dependabot-go-mod-tidy-and-vendor-support/

**Comment by tenzen-y** [comment]
> It looks like the builds are between 20 to 30s faster.
> 
> @tenzen-y wdyt?

I understand the valuables of the vendor.

**Comment by alculquicondor** [comment]
/retest
Flaky #2429

**Pull request merged**